modules/seqexec/server/src/main/scala/seqexec/server/InstrumentSystem.scala

```diff
@@ -29,7 +30,7 @@ trait InstrumentSystem[F[_]] extends System[F] with InstrumentGuide {
 
   def observeProgress(total: Time, elapsed: InstrumentSystem.ElapsedTime): Stream[F, Progress]
 
-  def instrumentActions(config: Config): InstrumentActions[F]
+  def instrumentActions(config: Config, tcsO: Option[Tcs[F]]): InstrumentActions[F]
```

Comment:
Could we instead pass `ObserveEnvironment[F]`. it contains both `config` and `systems: List[System[F]]` from where you can find `Tcs[F]`

Reply:
That is not a bad idea, and having all the systems involved in the step may be useful for future features.